db_sql.go

```diff
@@ -469,89 +539,155 @@ func SqlCreateDatabase(conx *DBConnection) error {
+	oldVersionTables := []string{"authgroup_version", "authsession_version", "authuser_version"}
+	// We first want to see if the new migrator has taken over, by checking if old version tracking tables exist
+	tableNameResult := ""
+	query := db.QueryRow(fmt.Sprintf("SELECT table_name FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_NAME = '%v'", oldVersionTables[0]))
+	err := query.Scan(&tableNameResult)
+	if err != nil {
+		if strings.Index(err.Error(), "no rows in result set") != -1 {
+			return true, nil
 		}
+		return false, err
+	}
+	// Tracking tables dont exist, this is up to date
+	if tableNameResult == "" {
+		return true, nil
 	}
-}
 
-// Create a Postgres DB schema necessary for a Session database
-func SqlCreateSchema_Session(conx *DBConnection) error {
-	versions := make([]string, 0)
-	versions = append(versions, `
-	CREATE TABLE authsession (id BIGSERIAL PRIMARY KEY, sessionkey VARCHAR, identity VARCHAR, permit VARCHAR, expires TIMESTAMP);
-	CREATE UNIQUE INDEX idx_authsession_token    ON authsession (sessionkey);
-	CREATE        INDEX idx_authsession_identity ON authsession (identity);
-	CREATE        INDEX idx_authsession_expires  ON authsession (expires);`)
-	// we made some changes that are not backwards compatible, need to force a logout for all users
-	versions = append(versions, `DELETE FROM authsession;`)
+	//
+	checkTableVersion := func(table string, expectedVersion int) bool {
+		var version int
+		row := db.QueryRow(fmt.Sprintf("SELECT version FROM %v", table))
+		err := row.Scan(&version)
+		if err != nil {
+			return false
+		} else {
+			return true
+		}
 
-	return MigrateSchema(conx, "authsession", versions)
-}
+		if version == expectedVersion {
+			return true
+		}
 
-// Create a Postgres DB schema suitable for storage of Permits and Authentication
-// Note that we COULD separate Permit and Authenticator, but I'm not sure that provides any value.
-// It would be trivial to do so if that need ever arises.
-func SqlCreateSchema_User(conx *DBConnection) error {
-	versions := make([]string, 0)
-	versions = append(versions, `
-	CREATE TABLE authuser (id BIGSERIAL PRIMARY KEY, identity VARCHAR, password VARCHAR, permit VARCHAR);
-	CREATE UNIQUE INDEX idx_authuser_identity ON authuser (identity);`)
+		return false
+	}
 
-	versions = append(versions, `
-	DROP INDEX idx_authuser_identity;
-	CREATE UNIQUE INDEX idx_authuser_identity ON authuser (LOWER(identity));`)
+	if checkTableVersion(oldVersionTables[0], 1) && checkTableVersion(oldVersionTables[1], 2) && checkTableVersion(oldVersionTables[2], 3) {
+		// Tracking tables do exist, we need to remove them
+		for _, oldVersionTable := range oldVersionTables {
+			_, err := db.Exec(fmt.Sprintf("DROP TABLE IF EXISTS %v", oldVersionTable))
+			if err != nil {
+				return false, err
+			}
+		}
+	}
 
-	versions = append(versions, `
-	ALTER TABLE authuser ADD COLUMN pwdtoken VARCHAR;`)
+	// Now we need to set the version in new migration version tracking table
```

Comment:
Generally avoid starting a comment with "Now....". It's generally a statement by a programmer of the code that needs to be written next, but a comment should read as though it's documenting the final product.
Also - no need to apologise for the way we're bootstrapping the burntsushi migration system. It's the only reasonable thing to do. We certainly do need an explanation in here, but don't make it sound as though there's a better solution.